sdk/src/types/block/semantic/mod.rs

```diff
@@ -105,17 +110,18 @@ impl<'a> SemanticValidationContext<'a> {
 
     ///
     pub fn validate(mut self) -> Result<Option<TransactionFailureReason>, Error> {
-        // Validation of inputs.
-        let mut has_implicit_account_creation_address = false;
-
-        self.commitment_context_input = self.transaction.context_inputs().commitment().copied();
+        self.commitment_context_input = self
```

Comment:
Seems weird to even store this in a field, since it's so easy to retrieve from the transaction

Reply:
It's really extra verbose if we don't :/

Reply:
It's used in a lot of state transitions as well